src/selection/useRange.tsx

```diff
@@ -43,20 +43,11 @@ export function useRange<T extends DayPickerProps>(
       ? addToRange(triggerDate, selected, min, max, required, dateLib)
       : undefined;
 
-    if (newRange?.from && newRange.to) {
-      let newDate = newRange.from;
-      while (dateLib.differenceInCalendarDays(newRange.to, newDate) > 0) {
-        newDate = dateLib.addDays(newDate, 1);
-        if (
-          excludeDisabled &&
-          disabled &&
-          dateMatchModifiers(newDate, disabled, dateLib)
-        ) {
-          // if a disabled days is found, the range is reset
-          newRange.from = triggerDate;
-          newRange.to = undefined;
-          break;
-        }
+    if (excludeDisabled && disabled && newRange?.from && newRange.to) {
+      if (rangeMatchModifiers(newRange, disabled, dateLib)) {
```

Comment:
`rangeMatchModifiers` also checks `newRange.from` and `newRange.to`. We could force the function to work only with `Required<DateRange>`, and remove the same checks in `rangeMatchModifiers`.